CMakeLists.txt

```diff
@@ -13,14 +13,19 @@ project(iree-bare-metal-arm VERSION 1.0 LANGUAGES C ASM)
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
 option(BUILD_WITH_CMSIS "Builds the sample with CMSIS" OFF)
+option(BUILD_WITH_CRT0 "Builds the sample with crt0.s" OFF)
 option(BUILD_WITH_LIBOPENCM3 "Builds the sample with libopencm3" OFF)
 
-if(BUILD_WITH_CMSIS AND BUILD_WITH_LIBOPENCM3)
-  message(FATAL_ERROR "libopencm3 and CMSIS cannot be used at the same time")
+if(
+  (BUILD_WITH_CMSIS AND BUILD_WITH_CRT0) OR
+  (BUILD_WITH_CMSIS AND BUILD_WITH_LIBOPENCM3) OR
+  (BUILD_WITH_CRT0 AND BUILD_WITH_LIBOPENCM3)
```

Comment:
We should refactor this in a follow up. Maybe a HAL/BSP_TYPE option with allows different values. But that is something for a followup.